### ISSUE TITLE
test_stream fail due to ZMQStream

### ISSUE THREAD

**Issue by scottkmaxwell** [circus-tent/circus#584]
If I run the tests in test_stream, the tests succeed but the client is actually failing with this exception:

Traceback (most recent call last):
  File "/Users/scottmax/Source/circus/circus/tests/support.py", line 87, in setUp
    self.cli = AsyncCircusClient()
  File "/Users/scottmax/Source/circus/circus/client.py", line 42, in **init**
    self.stream = ZMQStream(self.socket, tornado.ioloop.IOLoop.instance())
  File "/Users/scottmax/Source/circus/env27/lib/python2.7/site-packages/zmq/eventloop/zmqstream.py", line 101, in **init**
    self._init_io_state()
  File "/Users/scottmax/Source/circus/env27/lib/python2.7/site-packages/zmq/eventloop/zmqstream.py", line 522, in _init_io_state
    self.io_loop.add_handler(self.socket, self._handle_events, self._state)
  File "/Users/scottmax/Source/circus/env27/lib/python2.7/site-packages/tornado/ioloop.py", line 526, in add_handler
    self._impl.register(fd, events | self.ERROR)
  File "/Users/scottmax/Source/circus/env27/lib/python2.7/site-packages/tornado/platform/kqueue.py", line 41, in register
    self._control(fd, events, select.KQ_EV_ADD)
  File "/Users/scottmax/Source/circus/env27/lib/python2.7/site-packages/tornado/platform/kqueue.py", line 60, in _control
    fd, filter=select.KQ_FILTER_READ, flags=flags))
TypeError: argument must be an int, or have a fileno() method.

I traced through the code and the problem is that a ZMQ socket is being sent to the io_loop, which is actually expecting either an int or an object with a fileno() method. I tried creating a wrapper to add a fileno() method that returns socket.FD, but it turns out that other parts of the io_loop code fail to call the method, so we get KeyErrors. I tried modifying the code for ZMQStream to pass socket.FD instead of socket when it hooks up the handlers, and that seems pretty close to working. But some tests complain that a file descriptor is being reused.

Bottom line is that it really looks to me like there is either a bug in ZMQStream or there is a mismatch between the ZMQStream in pyzmq 13.1.0 and the io_loop in tornado 3.1.1.
